lib/utilities/collectResponseHeaders.js

```diff
@@ -75,6 +75,17 @@ function collectResponseHeaders(objectMD, corsHeaders, versioningCfg,
         responseMetaHeaders['x-amz-tagging-count'] =
           Object.keys(objectMD.tags).length;
     }
+    if (objectMD.retentionInfo && (objectMD.retentionInfo.retainUntilDate
+        && objectMD.retentionInfo.mode)) {
+            responseMetaHeaders['x-amz-object-lock-retain-until-date']
```

Comment:
Yeah I agree about this, eslint will definitely complain but I'll see if there's anything that can be done to make this look better - either bypassing lint or I will think of something else.

Reply:
What I did when I ran into this was to move the conditional to a variable, for example,
```js
const objectLocked = objectMD.retentionInfo && objectMD.retentionInfo.retainUntilDate
        && objectMD.retentionInfo.mode;
if (objectLocked) {
  // do the logic
}
```

Reply:
I am a huge fan that bools should always represent the "positive" form. It overall simplifies the logic and terms. The positive form is not always the default but generally is. For example, even though all this logic has to do with the objects being locked. The bool that dictates if the object is in fact locked might want to be false if it is. Something like `objectOpen` or `objectEditable` ( kinda guessing since I do not know any more terms here that might be more standard ). Note that "unlocked" is not a good name as its effectively a double negative. 

Consider a variable that dictates if a user can go through a portal ( some game idk ).
```js
if (!portalClosed) {
  // go in
}
// OR
if (portalOpen) {
  // go in
}
```

The first says we can go in the portal if it is not closed. Or the second is just it is open. The positive form here is using the portal for portal things.

This is a little difficult at times for conditions that are sort of "blocker" conditions. Just as in this case the purpose of a lock is to block something from happening. If a door had a lock and a deadbolt we might consider the lock "engaged" to be the lock would block the door from opening. So the door could be opened if all locks are not engaged. However, since the primary function of an actual lock is to prevent things from happening its positive value is true if the lock is engaged.

```javascript
const canOpenDoor = !deadbolt.engaged && !lock.engaged;
if (canOpenDoor) {
  // do good things
}
// or
const doorLocked = deadbolt.engaged || lock.engaged;
if (!doorLocked) {
  // double negative
}
```

Also prefixing booleans with is, can, has or similar word is also very powerful as it makes conditions turn almost into sentences. Compare "if I can open the door then open the door" to "if not door locked open door". 

tl;dr use positive forms for booleans, rename locked to something like open or writable or something in the positive form. 

This might be a lot of changes since so much work was put in to add "locking" but just like mentioning this sort of stuff. I may also have some terms wrong and the lock here is actually the positive case for something in which case carry on...

Reply:
+1